core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -93,6 +95,9 @@ object DockerContainer {
       network) ++
       environmentArgs ++
       dnsServers.flatMap(d => Seq("--dns", d)) ++
+      dnsSearch.flatMap(d => Seq("--dns-search", d)) ++
+      // NOTE: --dns-option on modern versions of docker, but is --dns-opt on docker 1.12
+      dnsOptions.flatMap(d => Seq("--dns-opt", d)) ++
```

Comment:
We need to be very careful here. While I understand that OW officially supports Docker 1.12, I know that higher Docker versions are used as well.

I'm not in favour of using the `--dns-opt` option here although we already know it won't work in higher versions. This is weak point that will definitely break in future or when using a higher Docker version today.

Reply:
It annoyingly needs to match the version of the docker client embedded in the invoker Dockerfile.  I could add a comment to the Dockerfile with a pointer to change this line of code when the version is (finally) updated.  I thought about making this more parameterized, but that seemed like overkill.

I do need to get the ndots:5 option into the user actions resolv.conf file to enable user actions to invoke kube services, so we do need to do something.